proof/archive.go

```diff
@@ -449,3 +449,55 @@ func (m *MultiArchiver) RemoveSubscriber(
 // A compile-time assertion to make sure MultiArchiver satisfies the
 // NotifyArchiver interface.
 var _ NotifyArchiver = (*MultiArchiver)(nil)
+
+// ReplaceLastProofInBlob attempts to replace the last proof in the blob with
+// the passed proof. This is useful when we want to update the proof with a new
+// one after a re-org.
+func ReplaceLastProofInBlob(ctx context.Context, p *Proof,
+	archive Archiver, headerVerifier HeaderVerifier) error {
+
+	assetID := p.Asset.ID()
+	scriptPubKey := p.Asset.ScriptKey.PubKey
+	locator := Locator{
+		AssetID:   &assetID,
+		ScriptKey: *scriptPubKey,
+	}
+
+	blob, err := archive.FetchProof(ctx, locator)
+	if err != nil {
+		return fmt.Errorf("unable to fetch current proof: %w", err)
+	}
+
+	f := &File{}
+	if err := f.Decode(bytes.NewReader(blob)); err != nil {
```

Comment:
In the future, it would be ideal to be able to do this _without_ reading the entire file in. IIUC, this is made a bit more difficult since we store the number of proofs as a prefix. However, I think we could do something like:
  * read the number of proofs
  * append the new proof and update the incremental hash 
  * seek into the file (depends on what type of `io.Reader` we actually have) to update the count of the number of proofs

Reply:
Yes, definitely. I thought we already had an issue, but couldn't find it. Will create a new one.

Reply:
SGTM, I think once we kick off the load tests (cc @positiveblue) this'll start to pop up in a few places.